algorithms/abstract_agent.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""Abstract Agent used for all agents.
+
+- Author: Curt Park
+- Contact: curt.park@medipixel.io
+"""
+
+import argparse
+import os
+from abc import ABC, abstractmethod
+
+import git
+import gym
+import torch
+
+
+class AbstractAgent(ABC):
+    """Abstract Agent used for all agents.
+
+    Args:
+        env (gym.Env): openAI Gym environment with discrete action space
+        args (argparse.Namespace): arguments including hyperparameters and training settings
+
+    Attributes:
+        env (gym.Env): openAI Gym environment with discrete action space
+        args (argparse.Namespace): arguments including hyperparameters and training settings
+        state_dim (int): dimension of state space
+        action_dim (int): dimension of action space
+        action_low (float): lower bound of the action value
+        action_high (float): upper bound of the action value
+
+    """
+
+    def __init__(self, env: gym.Env, args: argparse.Namespace):
+        """Initialization."""
+        self.args = args
+        self.env = env
+        self.state_dim = env.observation_space.shape[0]
+        self.action_dim = env.action_space.shape[0]
```

Comment:
I correct the mistake that was previously `self.action_dim = env.observation_space.shape[0]` to `self.action_dim = env.action_space.shape[0]`.